tests/rptest/tests/rbac_test.py

```diff
@@ -204,3 +205,36 @@ def test_get_role(self):
         assert role_members == [], f"Unexpected role members: {role_members} != {[]}"
 
         # TODO: add tests for role membership once assigning members to roles gets implemented
+
+    @cluster(num_nodes=3)
+    def test_update_role(self):
+        with expect_role_error(RoleErrorCode.ROLE_NOT_FOUND):
+            self.superuser_admin.update_role(
+                role=self.role_name0, update=RoleUpdate(role=self.role_name1))
+
+        self._create_and_wait_for_role(role=self.role_name0)
+
+        res = self.superuser_admin.update_role(
+            role=self.role_name0, update=RoleUpdate(role=self.role_name1))
+        new_role = res.json()['role']
+        assert new_role == self.role_name1, f"Unexpected role name: {new_role} != {self.role_name1}"
+
+        wait_until(lambda: self._role_exists(self.role_name1),
+                   timeout_sec=10,
+                   backoff_sec=2,
+                   err_msg="Updated (newly added) role does not exist")
+        wait_until(lambda: not self._role_exists(self.role_name0),
+                   timeout_sec=10,
+                   backoff_sec=2,
+                   err_msg="Updated (removed) role still exists")
```

Comment:
Yeah, I think it makes sense to capture that. I'll make that change.

I am wondering about 2 things though:
1. Is there any guarantee with ducktape that we'll hit the same node and same shard during a test? Or is that just highly likely?
2. As far as I can tell, the reads to the role store are only eventually consistent and there could be nodes lagging because we don't have a linearizable barrier/controller command replication on reads. I think that's fine in this case, but I'm wondering how we decide in general which read endpoints need a linearizable barrier.

Reply:
>how we decide

I was wondering that yesterday myself. Seems not to be emphasized on the admin API in general.

To the body of the test itself, all I mean is that two GET requests might, as you said, hit two different nodes. But since the  rename operation is "atomic", the response from a single GET should contain the new name iff of the old name has been overwritten. Probably not critical

Reply:
Yeah, I think that's what I've done now with `update_completes_atomically`, but let me know if you meant something different.

Reply:
We generally default to accepting eventually consistent behavior, and introduce stronger consistency only when needed.